src/operator/tensor/broadcast_reduce_op_index.cu

```diff
@@ -43,5 +43,18 @@ NNVM_REGISTER_OP(pick)
 NNVM_REGISTER_OP(_backward_pick)
 .set_attr<FCompute>("FCompute<gpu>", PickOpBackward<gpu>);
 
+template<>
+int DefineNumbWorkers<gpu>(const TShape &shape, int axis) {
+  const auto nSteps = static_cast<uint32_t>(shape[axis]);
+  const auto nThreads = shape.Size()/nSteps;
+  if (nThreads > nSteps)
+    return 1;
+
+  const auto a = static_cast<float>(nSteps)/nThreads;
+  const auto b = log2f(a);
+  const auto numbWorkers = pow(2, (b * 5 + 28)/11);
```

Comment:
Why is the number of workers pow(2, (b * 5 + 28)/11)?

Reply:
This is just a heuristic. Experimenting with different values `(shape, axis, numbWorkers)`, I accumulated a lot of data, it turned out that they almost perfectly corresponded to this formula.

Reply:
Could you add a line of comment about this heuristic?

Reply:
Is this only applicable to argmax? or it can be generalized?